### PR TITLE
Untested port of cis-benchmark action to the kubernetes-control-plane

### PR DESCRIPTION
**Pull request by addyess**
### Overview
Addresses [LP#2044219](https://bugs.launchpad.net/charm-kubernetes-master/+bug/2044219)

Re adds the cis-benchmark action from the reactive charm implementation

### Details
Takes a shot at porting the action from [layer-cis-benchmark](https://github.com/charmed-kubernetes/layer-cis-benchmark/blob/main/actions/cis-benchmark) into ops.  If it works out well -- likely convert to an ops version of this file in the original layer and included it as a pip package.

### CODE REVIEW

**Comment by VariableDeclared** [comment]
any chance of movement on this?

File: src/actions/csi_benchmark.py
Comment (bschimke95) [inline src/actions/csi_benchmark.py:1]: I guess the file name should be `csi_benchmark.py`, right?

File: src/actions/csi_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+    def run(self, test_num: int, test_remediation: str) -> int:
+        """Run the remedy command."""
+        if self.type == "manual":
+            log.info(
+                "Test %s: unable to auto-apply remedy.\n" "Manual steps:\n%s",
+                test_num,
+                test_remediation,
+            )
+        elif self.type == "cli":
```
Comment (bschimke95) [inline src/actions/csi_benchmark.py:49]: According to the comment above "kv" is also a possible type. Don't we need to handle this?

File: src/actions/csi_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+log = logging.getLogger(__name__)
+
+BENCH_HOME = Path("/home/ubuntu/kube-bench")
+BENCH_BIN = BENCH_HOME / "kube-bench"
+BENCH_CFG = BENCH_HOME / "cfg-ck"
+GO_PKG = "github.com/aquasecurity/kube-bench"
+RESULTS_DIR = "/home/ubuntu/kube-bench-results"
+
+# Remediation dicts associate a failing test with a Remedy to fix it.
```
Comment (bschimke95) [inline src/actions/csi_benchmark.py:24]: nit: should this comment be part of the dataclass `Remedy`?

File: src/actions/csi_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+}
+ADMISSION_PLUGINS = {
+    "enable-admission-plugins": (
+        "PersistentVolumeLabel",
+        "PodSecurityPolicy," "AlwaysPullImages",
+        "NodeRestriction",
+    )
+}
+DANGEROUS = {
+    "0.0.0": Remedy("cli", 'echo "this is fine"', None),
```
Comment (bschimke95) [inline src/actions/csi_benchmark.py:74]: Why do we have those dummy values in CONSERVATIVE and here?

File: src/actions/csi_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+        for name in files:
+            if name == filename:
+                log.info("Moving %s to %s", root, dest)
+                shutil.move(root, dest)
+                return
+    else:
+        raise ActionError(f"Could not find {filename} in {dirpath}")
+
+
+class CSIBenchmark(ops.Object):
```
Comment (bschimke95) [inline src/actions/csi_benchmark.py:135]: nit: CSI and in comment it is CIS

**Review by bschimke95 — COMMENTED** [review]
Some code comments. 
I cannot say much to the reactive (?) charm structure but I trust you that this is makes sense

File: src/actions/csi_benchmark.py (hunk 1 above)
Comment (addyess) [reply src/actions/csi_benchmark.py:49]: it doesn't look like the code path deals with that.  see https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/pull/349/files#diff-28032d63f14eb463321b6f903bc8a8ed4f8ee3e76991509f56912ad7157efed7R238-R244

File: src/actions/csi_benchmark.py (hunk 3 above)
Comment (addyess) [reply src/actions/csi_benchmark.py:74]: I don't know -- this comes from the original code
https://github.com/charmed-kubernetes/layer-cis-benchmark/blob/main/actions/cis-benchmark#L45

Comment (addyess) [reply src/actions/csi_benchmark.py:74]: these seemingly unnecessary values were originally introduced https://github.com/charmed-kubernetes/layer-cis-benchmark/pull/1

**Review by bschimke95 — APPROVED** [review]
LGTM

**Comment by addyess** [comment]
@VariableDeclared should be in 1.31/edge tomorrow... i'll kick the build to be sure

**Pull request merged**